src/qilib/configuration_helper/instrument_adapter_factory.py

```diff
@@ -69,6 +72,7 @@ def get_instrument_adapter(cls, instrument_adapter_class_name: str, address: str
         if adapter is None:
             raise ValueError(f"No such InstrumentAdapter {instrument_adapter_class_name}")
         else:
-            adapter = cast(InstrumentAdapter, adapter(address))
+            args = (address, instrument_name) if instrument_name is not None else address,
```

Comment:
```suggestion
            args = (address, instrument_name) if instrument_name is not None else (address,)
```
Quite subtle bug, apparently the `if-else` takes precedence over the `,`-tuple operator